src/Engine/Renderer/Mesh/Mesh.inl

```diff
@@ -464,10 +467,83 @@ PointCloud::PointCloud( const std::string& name,
 }
 
 /////////  LineMesh ///////////
+
 LineMesh::LineMesh( const std::string& name,
                     typename base::CoreGeometry&& geom,
                     typename base::MeshRenderMode renderMode ) :
     base( name, std::move( geom ), renderMode ) {}
 
+/////////  PolyMesh ///////////
+
+void PolyMesh::updateGL_specific_impl() {
+    if ( !m_indices )
+    {
+        m_indices      = globjects::Buffer::create();
+        m_indicesDirty = true;
+    }
+    if ( m_indicesDirty )
+    {
+        triangulate();
+        /// this one do not work since m_indices is not a std::vector
+        // m_indices->setData( m_mesh.m_indices, GL_DYNAMIC_DRAW );
+        m_numElements = m_triangleIndices.size() * Core::Vector3ui::RowsAtCompileTime;
+
+        m_indices->setData(
+            static_cast<gl::GLsizeiptr>( m_triangleIndices.size() *
+                                         sizeof( typename base::CoreGeometry::IndexType ) ),
+            m_triangleIndices.data(),
+            GL_STATIC_DRAW );
+        m_indicesDirty = false;
+    }
+    if ( !base::m_vao ) { base::m_vao = globjects::VertexArray::create(); }
+    base::m_vao->bind();
+    base::m_vao->bindElementBuffer( m_indices.get() );
+    base::m_vao->unbind();
+}
+
+void PolyMesh::triangulate() {
+    m_triangleIndices.clear();
+    m_triangleIndices.reserve( m_mesh.getIndices().size() );
+    for ( const auto& face : m_mesh.getIndices() )
+    {
+        LOG( logDEBUG ) << face.cols() << " " << face.rows() << " " << face.size();
+
+        if ( face.size() == 3 ) { m_triangleIndices.push_back( face ); }
+        else
+        {
```

Comment:
Note that both algorithms might fail if some vertices are aligned. Imagine the following rectangular element (where `x` denote a vertex, , the first one being identified as `v0`): 
```
   x---------------------------------------------------------------x
   |                                                               |
   |                                                               |
   x------x---------x----------x---------x-----------x-------------x
v0
```
both version will fail to generate usefull (even non manifold in some cases) triangulations. 
 - Fan will generate degenerates triangles (the first ones are degenerates)
 - Sew will generate overlapping triangle (from the 4th to the last)

Even if this case is very rare, there are others situations of aligned vertices, more frequent, that might fail the same way.

But for a first implementation, that will allow to test the overall behavior, it is fine for me.

Note that, event if triangulating a convex polygon seems simple, there is a huge litterature about how to do that robustly and with minimal complexity as the number of different triangulation of a n-vertex polygon is given by the (n-2) Catalan number (Euler). For the above exemple, with 9 vertices, there is 132 different triangulations (https://en.wikipedia.org/wiki/Polygon_triangulation)

Reply:
I agree, who cares about such malformed polygons ;)

Reply:
well, I'll switch back to the triangle fan implementation.
Here the purpose is to handle quad to triangle conversion on the fly, not to triangulate nicely the mesh.
If you want to do better, use openmesh's, or assimp's functionality. 
As said in the wikipedia page you link "It is trivial to triangulate any convex polygon in linear time into a fan triangulation, by adding diagonals from one vertex to all other vertices."
If you want to spend time on the PR, could you please focus on the PolyMesh to TopoMesh conversion and tests ?

Reply:
(BTW could you draw how sew creates overlapping triangle, on my side I found degenerated, but no overlapping)

Reply:
What will be the impact of these degenerated triangles in the rendering ?

Reply:
(And also: 99% of our cases will planar quads, so we don’t really care).

Reply:
sew (strip) is nice for $99.\bar{9}$ of our triangle, we'll keep this.
An yes, I'll focus on PolyMesh to TopoMesh conversion ;)